tests/h/cli/commands/user_test.py

```diff
@@ -62,266 +58,190 @@ def test_it_adds_user_with_specific_authority(self, cli, cliconfig, signup_servi
+        user_password_service.update_password.assert_not_called()
 
     @pytest.fixture
-    def user(self, db_session, factories):
-        user = factories.User()
-        db_session.flush()
-        return user
+    def user(self, factories):
+        return factories.User()
 
 
 class TestDeleteUserCommand:
-    def test_it_deletes_user(self, cli, cliconfig, user, db_session):
-        result = cli.invoke(user_cli.delete, [user.username], obj=cliconfig)
+    def test_it_deletes_user(self, invoke_cli, user, user_delete_service):
+        result = invoke_cli(user_cli.delete, [user.username])
 
         assert not result.exit_code
-        assert not db_session.query(models.User).filter_by(id=user.id).count()
+        user_delete_service.delete.assert_called_once_with(user)
 
     def test_it_deletes_user_with_specific_authority(
-        self, cli, cliconfig, user, db_session
+        self, invoke_cli, user, user_delete_service
     ):
         user.authority = "partner.org"
-        db_session.flush()
 
-        result = cli.invoke(
-            user_cli.delete,
-            ["--authority", "partner.org", user.username],
-            obj=cliconfig,
+        result = invoke_cli(
+            user_cli.delete, ["--authority", "partner.org", user.username]
         )
 
         assert not result.exit_code
-        assert not db_session.query(models.User).filter_by(id=user.id).count()
+        user_delete_service.delete.assert_called_once_with(user)
 
     def test_it_errors_when_user_could_not_be_found(
-        self, cli, cliconfig, user, db_session
+        self, invoke_cli, user_delete_service
     ):
-        result = cli.invoke(user_cli.delete, [f"bogus_{user.username}"], obj=cliconfig)
+        result = invoke_cli(user_cli.delete, ["bogus_username"])
 
         assert result.exit_code == 1
-        assert db_session.query(models.User).filter_by(id=user.id).count() == 1
+        user_delete_service.delete.assert_not_called()
 
     def test_it_errors_when_user_with_specific_authority_could_not_be_found(
-        self, cli, cliconfig, user, db_session
+        self, invoke_cli, user, user_delete_service
     ):
-        result = cli.invoke(
-            user_cli.delete, ["--authority", "foo.com", user.username], obj=cliconfig
-        )
+        result = invoke_cli(user_cli.delete, ["--authority", "foo.com", user.username])
 
         assert result.exit_code == 1
-        assert db_session.query(models.User).filter_by(id=user.id).count() == 1
+        user_delete_service.delete.assert_not_called()
 
     @pytest.fixture
-    def user(self, db_session, factories):
-        user = factories.User()
-        db_session.flush()
-        return user
+    def user(self, factories):
+        return factories.User()
 
 
 @pytest.fixture
-def signup_service():
-    signup_service = mock.Mock(spec_set=["signup"])
-    return signup_service
-
-
-@pytest.fixture
-def hasher():
-    # Use a much faster hasher for testing purposes. DO NOT use as few as
```

Comment:
Hard to resit the temptation to copy test code to production

Reply:
I have a PR in the works...